### PR TITLE
Create useHWAccount hook

### LINKED ISSUE #4775: Create the useHWAccounts hook
### Description
UseHWAccounts is a hook that allows for  account management access to the HW account and it will manipulate accounts in order to add account names from local storage and make them similar to the account management format. The main tasks of UseHWAccounts include:

### Acceptance Criteria
- Reading the HW account list from HWManager through the events emitted.
- Add unit test

### Additional information
The UseHWAccounts get list of accounts witch is already merged and ready from the HW Provider or store 

```js


{
  "hw": {
    deviceId: "",
    model: "",
    brand: "",
  },
  "metadata": {
    "name": "masoud123",
    "pubkey": "a07e06d3d21bfcbd6e6cb158a0adc18b6ccc92db66dc8a323c7462af9c539fab",
    "path": "",
    "accountIndex": 1, //HW index
    “isHW”: true,	
    "address": "lskdgtenb76rf93bzd56cqn6ova46wfvoesbk4hnd",
    "creationTime": "2022-12-08T15:30:39.979Z"
  },
  "version": 1
}
```

### PR DESCRIPTION
**Pull request by ikem-legend**
### What was the problem?

This PR resolves #4775

### How was it solved?

- Creating the required hook

### How was it tested?

- Unit tests

### CODE REVIEW

File: src/modules/wallet/utils/hwManager.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const getAccountBundle = async (deviceId, network, offset) => {
     publicKeyList.push(publicKey);
```
Comment (oskarleonard) [inline src/modules/wallet/utils/hwManager.js]: I guess this file is just formatting?

**Review by oskarleonard — DISMISSED** [review]

File: src/modules/wallet/utils/hwManager.js (hunk 1 above)
Comment (ikem-legend) [reply src/modules/wallet/utils/hwManager.js]: @oskarleonard Yes it is. Just one line was removed and the rest are formatting changes

Comment (soroushm) [reply src/modules/wallet/utils/hwManager.js]: please revert all the changes on this file

File: src/redux/selectors.js [hunk 2]
```diff
@@ -14,18 +14,20 @@ const selectActiveTokenAccount = (state) => {
...
-const selectStaking = state => state.staking;
+  state.bookmarks[state.token.active].find((item) => item.address === address);
+const selectSettings = (state) => state.settings;
+const selectNetwork = (state) => state.network;
+const selectServiceUrl = (state) => state.network.networks?.LSK?.serviceUrl;
+const selectNetworkIdentifier = (state) => state.network.networks?.LSK?.networkIdentifier;
+const selectNetworkName = (state) => state.network.name;
+const selectActiveTokenNetwork = (state) => state.network.networks[state.token.active];
+const selectStaking = (state) => state.staking;
+const selectHWAccounts = (state) => state.settings.hardwareAccounts;
```
Comment (soroushm) [inline src/redux/selectors.js]: hardaware wallet its not part of the setting
```suggestion
const selectHWAccounts = (state) => state.harwareWallet.Accounts;
```

**Review by soroushm — CHANGES_REQUESTED** [review]

File: src/redux/selectors.js (hunk 2 above)
Comment (ikem-legend) [reply src/redux/selectors.js]: @soroushm Hardware wallet is part of settings as seen in `src/modules/settings/store/reducer.js:18`. You even proposed some code with it `const {hardwareAccounts} = store.settings` in https://github.com/LiskHQ/discussions/discussions/99

File: src/modules/wallet/utils/hwManager.js (hunk 1 above)
Comment (ikem-legend) [reply src/modules/wallet/utils/hwManager.js]: @soroushm I had to remove a line or two of dead code and the rest is formatting changes. So why do we need to revert all changes on the file?

File: src/redux/selectors.js (hunk 2 above)
Comment (soroushm) [reply src/redux/selectors.js]: the hw name is part of the setting witch we dont want to change it
the accounts it should be store in the hw domain under "/src/modules/hardwareWallet/store"

File: src/modules/wallet/utils/hwManager.js (hunk 1 above)
Comment (soroushm) [reply src/modules/wallet/utils/hwManager.js]: we dont want to toch any existing utils, create the utlist you need on 'src/modules/hardwareWallet/utils"

File: src/redux/selectors.js (hunk 2 above)
Comment (ikem-legend) [reply src/redux/selectors.js]: Fixed

File: src/modules/wallet/utils/hwManager.js (hunk 1 above)
Comment (ikem-legend) [reply src/modules/wallet/utils/hwManager.js]: Fixed

**Review by oskarleonard — APPROVED** [review]

**Review by soroushm — APPROVED** [review]

**Pull request merged**